pandoc/filter.py

```diff
@@ -0,0 +1,97 @@
+#! /usr/bin/env python
+# encoding: utf-8
+
+import logging
+from panflute import Link, run_filter, Para, Str, Header, MetaInlines, Space, debug, Div, HorizontalRule, \
+    BulletList, RawBlock, Span, Emph, Strong
+
+logging.basicConfig(level=10)
+
+version = None
+
+pb = RawBlock(u"<w:p><w:r><w:br w:type=\"page\" /></w:r></w:p>", format=u"openxml")
+
+
+def action(elem, doc):
```

Comment:
There is a more delicate solution for filtering.

My test data is: `[1, 'a', {}, [1, 'a', 2], (1, 2)]` and i want to:
1. filter by type;
2. if something is unknown, filter it "differently".

Default code:

```
def filter(e):
    if isinstance(e, int):
        return 'this is int'
    if isinstance(e, str):
        return 'this is str'
    if isinstance(e, dict):
        return 'this is dict'
    if isinstance(e, list) and isinstance(element[0], int):
        return 'this is list[int]'
    return 'this is some object'
```
This is not a bad design when we have a **"one-line"** logic.
However, if we have complex logic, this block becomes difficult to understand.

I suggest using the following approach:

```
def handle_int(e):
    return 'this is int'

def handle_str(e):
    return 'this is str'

def handle_dict(e):
    return 'this is dict'

def handle_list(e):
    if isinstance(e[0], int):
        return 'this is list[int]'

def handle_object(e):
    return 'this is some object'

switcher = {
    int: handle_int,
    str: handle_str,
    dict: handle_dict,
    list: handle_list
}

def handle_filter(e):
    return switcher.get(type(e), handle_object)(e)
```

Tests:
```
for e in test_data:
    print(filter(e) == handle_filter(e), handle_filter(e))
True this is int
True this is str
True this is dict
True this is list[int]
True this is some object
```

Reply:
fixed